src/com/tony/server/ResponseDeterminer.java

```diff
@@ -23,12 +30,37 @@ public Response determineResponse(Request request) {
             if(authenticator.requiresAuthentication(request)
                     && !authenticator.isAuthenticated(request)){
                 return new UnauthorizedResponse();
+            } else if(request.getHttpMethod().equals("PATCH")){
+                String filePath = ((PatchResponse)router.route(request)).getFilePath();
+                if(!fileEtagMatchesGivenEtag(filePath, request.getHeaders().get("If-Match"))) {
+                    return new PreconditionFailedResponse();
+                } else {
+                    return router.route(request);
```

Comment:
you're repeating this line, is there any way you could do this only once instead?